internal/xds/translator/translator.go

```diff
@@ -229,6 +218,18 @@ func findXdsCluster(tCtx *types.ResourceVersionTable, name string) *clusterv3.Cl
 	return nil
 }
 
+func addXdsCluster(tCtx *types.ResourceVersionTable, name string, destinations []*ir.RouteDestination, tSocket *corev3.TransportSocket, isHTTP2 bool, isStatic bool) {
```

Comment:
IMO, the last two parameters(`isHTTP2` `isStatic`) has a smell of bad code, so you need add comments everytime you call `addXdsCluster`.

Reply:
agree, we should add a `addClusterArgs{}` in a future PR

Reply:
can you open a ticket to follow up?

Reply:
raised https://github.com/envoyproxy/gateway/issues/1141